### DISCUSSION TITLE
Non-container deployment?

### DISCUSSION THREAD

**Discussion by joshuar** [gowebly/gowebly#112] [Q&A]
Hey folks, 

Awesome project. I was looking to use this for one of my own projects which is a service that runs on a Linux desktop. I was hoping to use gowebly to build a web UI for the tool. 

With regards to deployment, is it possible to deploy without using a container? For example, can I just build and install a single binary like 'ol Go allows, or will there be additional files that need to be installed alongside the binary?

**Comment by koddr** [comment]
Hi, @joshuar 

Thank you for these kind words! 😉 

Yes, you can create a Systemd service to allow your application to start/restart on its own. Here is an example service file (usually created in `/lib/systemd/system/gowebly.service` on your GNU/Linux machine):

```ini
[Unit]
Description = gowebly

[Service]
Type           = simple
User           = root
Group          = root
LimitNOFILE    = 4096
Restart        = always
RestartSec     = 5s
StandardOutput = append:/root/gowebly/errors.log
StandardError  = append:/root/gowebly/errors.log
ExecStart      = /root/project/<YOU PROJECT BINNARY NAME HERE>

[Install]
WantedBy = multi-user.target
```

After that, we just have to enable it and start the service using `systemctl`:

```shell
[root@server ~]$ systemctl enable gowebly.service
[root@server ~]$ systemctl start gowebly
```
